.travis/yapf.sh

```diff
@@ -7,19 +7,70 @@ set -eo pipefail
 builtin cd "$(dirname "${BASH_SOURCE:-$0}")"
 
 ROOT="$(git rev-parse --show-toplevel)"
-builtin cd "$ROOT"
-
-yapf \
-    --style "$ROOT/.style.yapf" \
-    --in-place --recursive --parallel \
-    --exclude 'python/ray/cloudpickle' \
-    --exclude 'python/ray/dataframe' \
-    --exclude 'python/ray/rllib' \
-    -- \
-    'test' 'python'
-
-if ! git diff --quiet; then
-    echo 'Reformatted staged files. Please review and stage the changes.'
+builtin cd "$ROOT" || exit 1
+
+YAPF_FLAGS=(
+    "--style $ROOT/.style.yapf"
+    '--in-place'
+    '--recursive'
+    '--parallel')
+
+YAPF_EXCLUDES=(
+    '--exclude python/ray/dataframe'
+    '--exclude python/ray/rllib'
+    '--exclude python/ray/cloudpickle'
+    '--exclude python/build'
+    '--exclude python/ray/pyarrow_files'
+    '--exclude python/ray/core/src/ray/gcs'
+    '--exclude python/ray/common/thirdparty')
+
+# Add the upstream branch if it doesn't exist
+if ! [[ -e "$ROOT/.git/refs/remotes/upstream" ]]; then
+    git remote add 'upstream' 'https://github.com/ray-project/ray.git'
+fi
+
+# Only fetch master, not whole tree.
+git fetch 'upstream' 'master'
+
+# Format specified files
+format() {
+    yapf "${YAPF_FLAGS[@]}" -- "$@"
+}
+
+# Format files that differ from main branch. Ignores dirs that are not slated
+# for autoformat yet.
+format_changed() {
+    if ! git diff --diff-filter=ACM --quiet --exit-code 'upstream/master' 'HEAD' -- '*.py' &>dev/null; then
+        git diff --name-only 'upstream/master' 'HEAD' -- '*.py' | xargs -P 5 \
```

Comment:
Shouldn't this be diffing against the mergebase? Have you checked what happens if master gets ahead of the current branch?